database/CoreTransaction.java

```diff
@@ -139,6 +142,7 @@ protected void notifyClosed() {
     abstract void delete();
 
     protected void closeResources() {
+        context.diagnosticTxn().finish();
```

Comment:
Let's add a quick comment on the PR itself saying why NoOpTransaction is a common pattern in tracing